### PR TITLE
Add support for custom executable sections

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

**Comment by jkotas** [comment]
Needed for https://github.com/dotnet/corert/pull/983.

@kyulee1 @russellhadley PTLA

File: lib/ObjWriter/objwriter.cpp [hunk 1]
```diff
@@ -266,25 +266,33 @@ extern "C" bool CreateDataSection(ObjectWriter *OW, const char *SectionName,
...
+    unsigned Characteristics = COFF::IMAGE_SCN_MEM_READ;
 
-    if (!IsReadOnly) {
-      Characteristics |= COFF::IMAGE_SCN_MEM_WRITE;
+    if (IsExecutable) {
+      Characteristics |= COFF::IMAGE_SCN_CNT_CODE | COFF::IMAGE_SCN_MEM_EXECUTE;
+    } else if (!IsReadOnly) {
+      Characteristics |= COFF::IMAGE_SCN_CNT_INITIALIZED_DATA | COFF::IMAGE_SCN_MEM_WRITE;
     }
+    else {
```
Comment (kyulee1) [inline lib/ObjWriter/objwriter.cpp]: There are a couple of places that do not meet format requirement including your previous commit.
Can you run "python utils\ccformat.py --untidy" and address the format?

Comment (jkotas) [inline lib/ObjWriter/objwriter.cpp]: I have fixed this one, but I am not able to figure out how to make `python utils\ccformat.py --untidy` work - it dumps really long diff that contains the whole file. If I try to run it with `--fix` argument,  it does nothing.

Comment (jkotas) [inline lib/ObjWriter/objwriter.cpp]: Ok, figured it out - clang-format.exe has to be on your PATH to make this work.

**Comment by kyulee1** [comment]
LGTM

**Pull request merged**